GoogleForCreators/web-stories-wp
# Page Templates: Save image for custom page templates

## Context



When saving custom page templates we want to create and upload an image of it so that we can display them in the list as an image instead of a "live" preview.

## Summary



## Relevant Technical Choices



## To-do



- [x] Reorganize / DRY code
- [x] Don't use fullbleedContainer, but a custom node for rendering a 640x853 image (or another specified dimension)

## User-facing changes



Saved templates should now be in the 2:3 size like the default templates. It should also render images instead of the element page as needed.

## Testing Instructions






This PR can be tested by following these steps:

1. Go to the editor
2. make some edits to a page
3. save the page as a page template
4. see that the page shows up in the page template
5. should also eventually be rendered as an image, but that might take some time to do depending on how many saved templates you have that need images generated for them.

## Reviews

### Does this PR have a security-related impact?



No

### Does this PR change what data or activity we track or use?



No

### Does this PR have a legal-related impact?



No

## Checklist




---



Fixes #9574
